packages/react/src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -337,7 +350,34 @@ const Panel: React.FC<SelectPanelProps> = ({
 }
 
 const SelectPanelButton = React.forwardRef<HTMLButtonElement, ButtonProps>((props, anchorRef) => {
-  return <Button ref={anchorRef} {...props} />
+  const inputProps = useFormControlForwardedProps(props)
+  const [labelId, setLabelId] = useState('')
+  useEffect(() => {
+    const label = document.querySelector(`[for='${inputProps.id}']`)
+    if (label) {
+      if (label.id) {
+        setLabelId(label.id)
+      } else {
+        const newLabelId = `${inputProps.id}--select-panel-button-label`
+        label.id = newLabelId
+        setLabelId(newLabelId)
+      }
+    }
+  }, [inputProps.id])
+
+  const selectPanelButtonId = `${inputProps.id}--select-panel-button`
+  if (inputProps.id && labelId) {
+    return (
+      <Button
+        ref={anchorRef}
+        aria-labelledby={`${labelId} ${selectPanelButtonId}`}
+        {...inputProps}
+        id={selectPanelButtonId}
+      />
+    )
+  } else {
+    return <Button ref={anchorRef} {...props} />
+  }
```

Comment:
The other implementation I tried was
```tsx
return (
    <>
      {inputProps.id && (
        <VisuallyHidden id={`${inputProps.id}--select-panel-button-children`}>{inputProps.children}</VisuallyHidden>
      )}
      <Button
        ref={anchorRef}
        {...inputProps}
        aria-labelledby={`${inputProps.id} ${inputProps.id}--select-panel-button-children`}
      />
    </>
  )
```
However I wasn't thrilled about using `VisuallyHidden` because it can have [unintended consequences](https://github.com/github/primer/issues/2865#issuecomment-1999910555), although it is used in the [discussion](https://github.com/primer/react/pull/4389) @siddharthkp referenced so it might not be problematic. 

The solution I ended up implementing is similar to what @TylerJDev suggested:
```html
<label id="label1">SelectPanel within FormControl</label>
<button id="label2" aria-labelledby="label1 label2">Choose a tag</button>
```

The only wrinkle is that I had to give the `Button` a new `id` (so it's no longer linked to the `for` tag in the `label`).

Is there a more elegant solution here, perhaps involving forwarding React refs?

Reply:
When I tried reusing the id that's generated by the FormControl for this purpose it ended up giving it the name: "SelectPanel within FormControl SelectPanel within FormControl" since the label overrides the inner text of the button when linking with the `for` attribute

Reply:
Sorry, I've been busy with first responder duties. I need to check if this is safe/fine.

@TylerJDev can you validate this for me, please?

Reply:
> When I tried reusing the id that's generated by the FormControl for this purpose it ended up giving it the name: "SelectPanel within FormControl SelectPanel within FormControl" 

I'm curious about where it gave it the name twice? Could we not utilize the ID `FormControl` provides, instead of using `selectPanelButtonId`? I see the potential issue where the `for` _could_ duplicate the label text, but when I tested, it stated both properly 🤔

Reply:
I'll take a video and sent it over. When I was doing it it was duplicating the text

Reply:
Oh interesting. 

When it's this order it works
```tsx
aria-labelledby={`${labelId} ${inputProps.id}`}
```

When it's this order it duplicates "SelectPanel within FormControl`
```tsx
aria-labelledby={`${inputProps.id} ${labelId}`}
```

https://github.com/primer/react/assets/29404461/3a61cd2c-b8a4-465e-9bf7-149af9741db2

Reply:
Alright I went ahead and removed `selectPanelButtonId` in favor of `inputProps.id` to avoid and a11y issues.